resources/aws_ec2.go

```diff
@@ -663,3 +714,365 @@ func (obj *AwsEc2Res) UnmarshalYAML(unmarshal func(interface{}) error) error {
 func (obj *AwsEc2Res) prependName() string {
 	return AwsPrefix + obj.GetName()
 }
+
+// snsServer returns an http server used to listen for sns messages
+func (obj *AwsEc2Res) snsServer() *http.Server {
+	port := fmt.Sprintf(":%d", int(obj.SnsPort))
+	srv := &http.Server{Addr: port}
+	log.Printf("%s: Starting SNS Endpoint", obj)
+	//rand.Seed(time.Now().UTC().UnixNano())
+	http.HandleFunc("/", obj.postHandler)
+	log.Printf("%s: Listening on %s", obj, port)
+	go func() {
+		if err := srv.ListenAndServe(); err != nil {
+			select {
+			case obj.awsChan <- &chanStruct{
+				err: errwrap.Wrapf(err, "error starting http server"),
+			}:
+			case <-obj.closeChan:
+			}
+			return
+		}
+	}()
+	return srv
+}
+
+// postHandler listens for posts on the SNS Endpoint
+// TODO: download pem and check message against signature
+func (obj *AwsEc2Res) postHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "POST" {
+		post, _ := ioutil.ReadAll(req.Body)
+		// unmarshall json
+		type data struct {
+			Type    string
+			Token   string
+			Message string
+		}
+		var p data
+		err := json.Unmarshal(post, &p)
+		if err != nil {
+			select {
+			case obj.awsChan <- &chanStruct{
+				err: errwrap.Wrapf(err, "error unmarshaling policy"),
+			}:
+			case <-obj.closeChan:
+			}
+			return
+		}
+		if p.Type == "SubscriptionConfirmation" {
```

Comment:
Is this a constant somewhere in the aws API?

Reply:
Nope. :( Not that I could find, anyway. I did find "sns" though.